common/overlay/overlay_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package overlay
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+	"testing"
+)
+
+// The tempDir creates a new directory with a given prefix. Panics if
+// creation of a directory results into the error.
+func tempDir(prefix string) string {
+	dir, err := ioutil.TempDir("", prefix)
+	if err != nil {
+		panic(err)
+	}
+
+	return dir
+}
+
+// The tempFile creates a new file within the given directory with a
+// specified name. Panics if file creation fails for any reason.
+func tempFile(dir, name string) string {
+	name = filepath.Join(dir, name)
+	file, err := os.OpenFile(name, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0600)
+	if err != nil {
+		panic(err)
+	}
+
+	defer file.Close()
+	return name
+}
+
+func testMount(t *testing.T, cfg MountCfg) error {
+	// Create a temporary directories with a configured prefix.
+
+	// Lower and Upper directories will create two text files
+	// to ensure the merging completed successfully.
+	cfg.Lower = tempDir(cfg.Lower)
+	tempFile(cfg.Lower, "1.txt")
+
+	cfg.Upper = tempDir(cfg.Upper)
+	tempFile(cfg.Upper, "2.txt")
+
+	cfg.Work = tempDir(cfg.Work)
+	cfg.Dest = tempDir(cfg.Dest)
+
+	// Ensure temporary directories will be removed after tests.
+	defer os.RemoveAll(cfg.Lower)
+	defer os.RemoveAll(cfg.Upper)
+	defer os.RemoveAll(cfg.Work)
+	defer os.RemoveAll(cfg.Dest)
+
+	err := Mount(&cfg)
+	if err != nil {
+		return err
+	}
+
+	// Unmount the destination directory when time comes.
+	defer syscall.Unmount(cfg.Dest, 0)
+
+	// Ensure both, 1.txt and 2.txt have been merged into the
+	// destination directory.
+	for _, ff := range []string{"1.txt", "2.txt"} {
+		_, err := os.Stat(filepath.Join(cfg.Dest, ff))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func TestMount(t *testing.T) {
+	if os.Geteuid() != 0 {
```

Comment:
Unfortunately, Go ```syscall``` wrapper does not allow (due to mentioned in the comments reason) to gain root privileges: [syscall/syscall_linux.go#L861](https://github.com/golang/go/blob/master/src/syscall/syscall_linux.go#L861). Link to the original issue: https://github.com/golang/go/issues/1435.

Therefore I didn't find anything smarter than run these tests only when they've been launched with superuser privileges.